contract/src/main/java/io/wax911/emojify/contract/model/AbstractEmoji.kt

```diff
@@ -0,0 +1,30 @@
+package io.wax911.emojify.contract.model
+
+import java.nio.charset.Charset
+
+abstract class AbstractEmoji: IEmoji {
+    val unicode: String by lazy {
+        String(emoji.toByteArray(), Charset.forName("UTF-8"))
+    }
+
+    val htmlDec by lazy {
+        html("&#%d;")
+    }
+
+    val htmlHex by lazy {
+        html("&#x%x;")
+    }
```

Comment:
I would rather override the properties and move them into `IEmoji` and substitute all references of `AbstractEmoji` with `IEmoji` in addition to disabling thread safety introduced by the `lazy` delegate: 

```suggestion
    override val unicode: String by lazy(LazyThreadSafetyMode.NONE) {
        String(emoji.toByteArray(), Charset.forName("UTF-8"))
    }

    override val htmlDec by lazy(LazyThreadSafetyMode.NONE) {
        html("&#%d;")
    }

    override val htmlHex by lazy(LazyThreadSafetyMode.NONE) {
        html("&#x%x;")
    }
```